pkgs/by-name/re/revolver/package.nix

```diff
@@ -0,0 +1,95 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchFromGitHub,
+  zsh,
+  installShellFiles,
+  ncurses,
+  nix-update-script,
+  testers,
+  runCommand,
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "revolver";
+  version = "0.2.4-unstable-2020-09-30";
+
+  src = fetchFromGitHub {
+    owner = "molovo";
+    repo = "revolver";
+    rev = "6424e6cb14da38dc5d7760573eb6ecb2438e9661";
+    hash = "sha256-2onqjtPIsgiEJj00oP5xXGkPZGQpGPVwcBOhmicqKcs=";
+  };
+
+  strictDeps = true;
+  dontConfigure = true;
+  dontBuild = true;
+  doInstallCheck = true;
+
+  nativeBuildInputs = [ installShellFiles ];
+  propagatedBuildInputs = [
+    zsh
+    ncurses
+  ];
```

Comment:
Could we easily patch the invocations of those commands to avoid having to propagate them?

Reply:
They are now `buildInputs`, with `tput` (from `ncurses`) patched with `substituteInPlace`.

At this point, however, I'm confused on the differences between `propagatedBuildInputs` and `buildInputs`.

Having these packages not propagated means that they are not added to `PATH` when downstream dependencies are installed? Am I missing something?